cluster-autoscaler/core/scale_down.go

```diff
@@ -1067,11 +1073,22 @@ func drainNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface
 		if allGone {
 			klog.V(1).Infof("All pods removed from %s", node.Name)
 			// Let the deferred function know there is no need for cleanup
-			return nil
+			return evictionResults, nil
 		}
 	}
-	return errors.NewAutoscalerError(
-		errors.TransientError, "Failed to drain node %s/%s: pods remaining after timeout", node.Namespace, node.Name)
+
+	for _, pod := range pods {
+		podReturned, err := client.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
+		if err == nil && (podReturned == nil || podReturned.Spec.NodeName == node.Name) {
+			evictionResults[pod.Name] = status.PodEvictionResult{Pod: pod, TimedOut: true, Err: nil}
+		} else if err != nil && !kube_errors.IsNotFound(err) {
+			evictionResults[pod.Name] = status.PodEvictionResult{Pod: pod, TimedOut: true, Err: err}
+		} else {
+			evictionResults[pod.Name] = status.PodEvictionResult{Pod: pod, TimedOut: false, Err: nil}
```

Comment:
nit: I'd prefer to gather errors where the original checks are made, instead of duplicating the code and introducing the new case which basically means "we succeeded after all, but it's too late".